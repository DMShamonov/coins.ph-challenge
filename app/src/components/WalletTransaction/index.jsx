// @flow

import * as React from 'react';

import { TableCell, TableRow } from 'components/Table';
import Value from 'components/Value';
import { calcBtc } from 'controllers/utilities';

import type { PropsType } from './types.js.flow';

export default function WalletTransaction({
  date,
  fee,
  value,
  pending,
}: PropsType): React.Node {
  return (
    <TableRow>
      <TableCell>{new Date(date).toISOString()}</TableCell>
      <TableCell width="10%">{fee}</TableCell>
      <TableCell width="10%">
        <Value value={calcBtc(value)} />
      </TableCell>
      <TableCell>{pending ? 'pending...' : 'success'}</TableCell>
    </TableRow>
  );
}
