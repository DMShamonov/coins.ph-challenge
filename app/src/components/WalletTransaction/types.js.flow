// @flow

export type PropsType = {|
  date: string,
  fee: number,
  value: number,
  pending: boolean,
|};
