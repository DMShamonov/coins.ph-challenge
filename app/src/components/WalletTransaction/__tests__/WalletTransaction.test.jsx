import React from 'react';
import { shallow } from 'enzyme';

import WalletTransaction from '../index';

describe('`WalletTransaction` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<WalletTransaction date="2019-04-09T02:56:04.152Z" fee={1000} value={100000000} pending={false} />)).toMatchSnapshot();
  });

  test('when `pending` prop is `true`', () => {
    expect(shallow(<WalletTransaction date="2019-04-09T02:56:04.152Z" fee={1000} value={100000000} pending />)).toMatchSnapshot();
  });
});
