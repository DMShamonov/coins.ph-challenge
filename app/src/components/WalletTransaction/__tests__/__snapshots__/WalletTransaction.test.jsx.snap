// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`WalletTransaction\` component should render correctly when \`pending\` prop is \`true\` 1`] = `
<TableRow>
  <TableCell>
    2019-04-09T02:56:04.152Z
  </TableCell>
  <TableCell
    width="10%"
  >
    1000
  </TableCell>
  <TableCell
    width="10%"
  >
    <Value
      value={1}
    />
  </TableCell>
  <TableCell>
    pending...
  </TableCell>
</TableRow>
`;

exports[`\`WalletTransaction\` component should render correctly with default props 1`] = `
<TableRow>
  <TableCell>
    2019-04-09T02:56:04.152Z
  </TableCell>
  <TableCell
    width="10%"
  >
    1000
  </TableCell>
  <TableCell
    width="10%"
  >
    <Value
      value={1}
    />
  </TableCell>
  <TableCell>
    success
  </TableCell>
</TableRow>
`;
